vim-patch:8.0.1748

janlazo opened the pull request:
**vim-patch:8.0.1748: CmdlineEnter command uses backslash instead of slash**

Problem:    CmdlineEnter command uses backslash instead of slash.
Solution:   Don't treat the character as a file name. (closes vim/vim#2837)
https://github.com/vim/vim/commit/a4baf5b32519855bb176a7aa0e9397c137ca890a

janlazo committed:
vim-patch:8.0.1748: CmdlineEnter command uses backslash instead of slash

Problem:    CmdlineEnter command uses backslash instead of slash.
Solution:   Don't treat the character as a file name. (closes vim/vim#2837)
https://github.com/vim/vim/commit/a4baf5b32519855bb176a7aa0e9397c137ca890a

janlazo commented:
```
From test_autocmd.vim:
Found errors in Test_Cmdline():
function RunTheTest[35]..Test_Cmdline line 20: Expected '/' but got 'C:\\'
function RunTheTest[35]..Test_Cmdline line 21: Expected '/' but got 'C:\\'
```

Didn't work

justinmk commented:
isn't this missing a change from the patch?

    	    autocmd_fname_full = TRUE; /* don't expand it later */

janlazo commented:
`autocmd_fname_full` was removed in 82cd0be2eaf71c0476e15c66ba3e83c76896d407
How do I add that change?

justinmk commented:
I think just add a special case for `/` [here](https://github.com/neovim/neovim/commit/82cd0be2eaf71c0476e15c66ba3e83c76896d407#diff-e5a85cda52db26f01de2cf7c1f2107cfR8550). It's not going to hurt anything, even on Windows.

```c
    case SPEC_AFILE:  // file name for autocommand
      if (autocmd_fname != NULL && !strequal("/", autocmd_fname) && !path_is_absolute_path(autocmd_fname)) {
```

janlazo committed:
ex_docmd: '/' is not a path for Cmdline* events

Code from https://github.com/neovim/neovim/pull/9348#issuecomment-446416118

autocmd_fname_full was removed in https://github.com/neovim/neovim/commit/82cd0be2eaf71c0476e15c66ba3e83c76896d407
but Vim uses this hack for <afile> on CmdlineEnter and related events
in vim-patch:8.0.1748.
Port the hack by not treating "/" as a path for <afile> on these events.

janlazo commented:
I don't see the error on Appveyor now.

justinmk commented on src/nvim/ex_docmd.c:
```diff
@@ -8654,7 +8654,10 @@ eval_vars (
+      // For CmdlineEnter and related events, <afile> is not a path! #9348
+      if (autocmd_fname != NULL
+          && !path_is_absolute(autocmd_fname)
+          && !strequal("/", (char *)autocmd_fname)) {
```
the comment should be attached to this line, it is not relevant to the other conditions
